includes/managers/class-fs-plan-manager.php

```diff
@@ -106,32 +106,59 @@ function has_free_plan( $plans ) {
 			return false;
 		}
 
-		/**
-		 * Find all plans that have trial.
-		 *
-		 * @author Vova Feldman (@svovaf)
-		 * @since  1.0.9
-		 *
-		 * @param FS_Plugin_Plan[] $plans
-		 *
-		 * @return FS_Plugin_Plan[]
-		 */
-		function get_trial_plans( $plans ) {
-			$trial_plans = array();
-
-			if ( is_array( $plans ) && 0 < count( $plans ) ) {
-				/**
-				 * @var FS_Plugin_Plan[] $plans
-				 */
-				for ( $i = 0, $len = count( $plans ); $i < $len; $i ++ ) {
-					if ( $plans[ $i ]->has_trial() ) {
-						$trial_plans[] = $plans[ $i ];
-					}
-				}
-			}
-
-			return $trial_plans;
-		}
+        /**
+         * Find all plans that have trial.
+         * Since 2.6.2 call get_filtered_plan
+         *
+         * @author Vova Feldman (@svovaf)
+         * @since  1.0.9
+         *
+         * @param FS_Plugin_Plan[] $plans
+         *
+         * @return FS_Plugin_Plan[]
+         */
+        function get_trial_plans( $plans ) {
+            return $this->get_filtered_plans($plans, true);
+        }
+
+        /**
+         * Find all plans that are not hidden and have trial.
+         *
+         * @author Daniele Alessandra (@danielealessandra)
+         * @since  2.6.3
+         *
+         * @param FS_Plugin_Plan[] $plans
+         *
+         * @return FS_Plugin_Plan[]
+         */
+        function get_visible_trial_plans( $plans ) {
+            return $this->get_filtered_plans($plans, true, true);
+        }
+
+        /**
+         * Find all plans filtered by trial or visibility.
+         *
+         * @author Daniele Alessandra (@danielealessandra)
+         * @since  2.6.3
+         *
+         * @param FS_Plugin_Plan[] $plans
+         *
+         * @return FS_Plugin_Plan[]
+         */
+        function get_filtered_plans($plans, $filter_trial = false, $filter_hidden = false) {
```

Comment:
@DanieleAlessandra the terms `$filter_trial` and `$filter_hidden` are doing the opposite things:
1. `$filter_trial` - Including only the plans which has trials
2. `$filter_hidden` - Excluding all hidden plans

So I propose to rename them to
1. `$should_have_trial`
2. `$should_be_visible`

or something like that.

Also please include the params in the PHP Doc block.